core/silkworm/execution/processor.cpp

```diff
@@ -66,9 +66,12 @@ ValidationResult ExecutionProcessor::validate_transaction(const Transaction& txn
     return ValidationResult::kOk;
 }
 
-Receipt ExecutionProcessor::execute_transaction(const Transaction& txn) noexcept {
+void ExecutionProcessor::execute_transaction(const Transaction& txn, Receipt& receipt) noexcept {
     assert(validate_transaction(txn) == ValidationResult::kOk);
 
+    // Optimization: since receipt.logs might have some capacity, let's reuse it.
+    std::swap(receipt.logs, state_.logs());
```

Comment:
`std::swap` is defined in vector so maybe semnatically is better `receipt.logs.swap(state_.logs());`

Reply:
I prefer `std::swap(a, b)` to highlight the symmetry of the operation, but it's a minor aesthetical point.

Reply:
Fair enough